mmdet/models/dense_heads/fcos_head.py

```diff
@@ -194,16 +208,24 @@ def loss(self,
             pos_decoded_bbox_preds = distance2bbox(pos_points, pos_bbox_preds)
             pos_decoded_target_preds = distance2bbox(pos_points,
                                                      pos_bbox_targets)
+            # average pos_centerness_targets from all gpus,
+            # which is used to normalize centerness-weighed reg loss
+            pos_centerness_targets_avg_per_gpu = \
+                reduce_mean(pos_centerness_targets.sum()).item()
+
             # centerness weighted iou loss
             loss_bbox = self.loss_bbox(
                 pos_decoded_bbox_preds,
                 pos_decoded_target_preds,
                 weight=pos_centerness_targets,
-                avg_factor=pos_centerness_targets.sum())
-            loss_centerness = self.loss_centerness(pos_centerness,
-                                                   pos_centerness_targets)
+                avg_factor=pos_centerness_targets_avg_per_gpu)
+            loss_centerness = self.loss_centerness(
+                pos_centerness,
+                pos_centerness_targets,
+                avg_factor=num_pos_avg_per_gpu)
         else:
             loss_bbox = pos_bbox_preds.sum()
+            reduce_mean(pos_centerness.new_tensor([0.0]))
```

Comment:
We might need to send 1 for reduce in case all GPUS have no positive sample?

Reply:
Well, I almost have no knowledge in DDP and just copied [that line](https://github.com/tianzhi0549/FCOS/blob/5fd728bfea150359fae28acb527e6e89d43c27e3/fcos_core/modeling/rpn/fcos/loss.py#L279) from the author of FCOS. So if you think it necessary to modify it, please feel free to do it.